From: Thomas Lange <lange@debian.org>
Date: Mon, 4 Aug 2025 13:10:56 +0200
Subject: udevsettle

In a KVM VM the network is not configured properly, it shows ens3
(instead of eth0), which is not in status up and no IP address.
---
 modules.d/80base/init.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/80base/init.sh b/modules.d/80base/init.sh
index fcbd4bb..47a8655 100755
--- a/modules.d/80base/init.sh
+++ b/modules.d/80base/init.sh
@@ -164,6 +164,7 @@ udevadm control --reload > /dev/null 2>&1 || :
 # then the rest
 udevadm trigger --type=subsystems --action=add > /dev/null 2>&1
 udevadm trigger --type=devices --action=add > /dev/null 2>&1
+udevadm settle
 
 make_trace_mem "hook initqueue" '1:shortmem' '2+:mem' '3+:slab'
 getargs 'rd.break=initqueue' && emergency_shell -n initqueue "Break before initqueue"
